File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/structure/ValidationRunSortingMode.kt
package net.nemerosa.ontrack.model.structure

/**
 * Defines how validation runs should be sorted.
 */
enum class ValidationRunSortingMode {

    /**
     * Sorting by decreasing ID (the default), therefore having the most recent validation in front.
     */
    ID,

    /**
     * Sorting by decreasing run time, therefore having the slowest validations in front.
     */
    RUN_TIME,

    /**
     * Sorting by alphabetical order.
     */
    NAME,

}

File: ontrack-web-core/components/validationRuns/ValidationRunSortingMode.js
import {Select} from "antd";

export default function ValidationRunSortingMode({value, onChange}) {
    return (
        <Select
            value={value}
            options={[
                {
                    value: "ID",
                    label: "From the newest"
                },
                {
                    value: "RUN_TIME",
                    label: "From the slowest"
                },
                {
                    value: "NAME",
                    label: "Alphabetical order"
                }
            ]}
            onChange={onChange}
        />
    )
}